packages/minimongo/selector.js

```diff
@@ -649,6 +690,108 @@ ELEMENT_OPERATORS = {
       };
     }
   },
+  $bitsAnyClear: {
+    compileElementSelector: function (operand, valueSelector, matcher) {
+      if (!isArray(operand) && !(operand instanceof Uint8Array))
+        throw Error("$bitsAnyClear has to be an Array");
+      return function (value) {
+        if (typeof value !== 'number' && !(value instanceof Uint8Array))
+          return false;
+
+        if (value === 0)
+          return true;
+
+        operand = ensureOperandUint8Array(operand);
+
+        if (!(value instanceof Uint8Array))
+          value = convertNumberToUint8Array(value);
+
+        return typeof _.find(operand, function (op, i) {
+            var bitsSetOp = get8BitsSet(op);
+            var bitsSetVal = get8BitsSet(value[i]);
+
+            return typeof _.find(bitsSetOp, function (bit) {
+                return bitsSetVal.indexOf(bit) === -1;
+              }) === 'undefined';
+          }) === 'undefined';
+      };
+    }
+  },
+  $bitsAllClear: {
+    compileElementSelector: function (operand, valueSelector, matcher) {
+      if (!isArray(operand) && !(operand instanceof Uint8Array))
+        throw Error("$bitsAllClear has to be an Array");
+      return function (value) {
+        if (typeof value !== 'number' && !(value instanceof Uint8Array))
+          return false;
+
+        if (value === 0)
+          return true;
+
+        operand = ensureOperandUint8Array(operand);
+
+        if (!(value instanceof Uint8Array))
+          value = convertNumberToUint8Array(value);
+
+        return _.filter(operand, function (op, i) {
+          var bitsSetOp = get8BitsSet(op);
+          var bitsSetVal = get8BitsSet(value[i]);
+
+          return typeof _.find(bitsSetOp, function (bit) {
+            return bitsSetVal.indexOf(bit) !== -1;
+          }) !== 'undefined';
+        }).length === 0;
+      };
+    }
+  },
+  $bitsAllSet: {
+    compileElementSelector: function (operand, valueSelector, matcher) {
+      if (!isArray(operand) && !(operand instanceof Uint8Array))
+        throw Error("$bitsAllSet has to be an Array");
+      return function (value) {
+        if (typeof value !== 'number' && !(value instanceof Uint8Array))
+          return false;
+
+        operand = ensureOperandUint8Array(operand);
+
+        if (!(value instanceof Uint8Array))
+          value = convertNumberToUint8Array(value);
```

Comment:
Could you change this to something like `ensureUint8Array` and push the `instanceof` check into the function, rather than doing it multiple times in each method?